morf-core/src/main/java/org/alfasoftware/morf/sql/element/TableReference.java

```diff
@@ -139,10 +139,10 @@ public TableReference(String tableName, boolean temporary) {
    * Specifies the alias to use for the table.
    *
    * @param aliasName the name of the alias
-   * @return an updated {@link TableReference} (this will not be a new object)
+   * @return an updated {@link TableReference} (this will not be a new object unless using immutable DSL option and the alias is being changed)
    */
   public TableReference as(String aliasName) {
-    if (AliasedField.immutableDslEnabled()) {
+    if (AliasedField.immutableDslEnabled() && !aliasName.equals(alias)) {

```

Comment:
Good change.
Generally, it could have lead to problems (usage relying on new object being created here),
however, this is the only mutating method on this class right now, and therefore no problems possible.